lib/failure_detector.ml

```diff
@@ -0,0 +1,138 @@
+open Base.Hashtbl
+
+type message =
+  | Ping
+  | Acknowledge
+  | PingRequest of Address.t
+[@@deriving bin_io]
+
+type config = {
+  protocol_period : int;
+  round_trip_time : int;
+  timeout : int;
+  mutable peers_to_ping : int;
```

Comment:
This field doesn't have to be mutable, we can just make the config field in `t` mutable. Then, if we want to change the number, we can do:

```ocaml
let failure_detector : Failure_detector.t = ... in
failure_detector.config <- { failure_detector.config with peers_to_ping = 5 };
```

Unless, for some reason, we want only this config value to be mutable?

Reply:
I got your point, but I am not fond of making the whole `type t` mutable when there is only one field which must be mutable.
What do you think?

Reply:
I don't think I understand why only this value is mutable 🤔

Reply:
Actually, after reading (again) the SWIM paper, I feel like this is not useful to make this field configurable from the client.
It looks like this number is "randomly decided by the protocol" and can be `internal`, what do you think?

Reply:
Sounds good; some things are supposed to be configurable though, right? What were those?

Reply:
Every field must be configurable, but I think it is better to "force" the client to create a new `failure_detector.t`.
Maybe we can expose a `create` function to ease this.
Do you think it is a good idea?